trillian/integration/hammer.go

```diff
@@ -1067,6 +1067,9 @@ func (s *hammerState) retryOneOp(ctx context.Context) error {
 	deadline := time.Now().Add(s.cfg.MaxRetryDuration)
 
 	for {
+		if err := ctx.Err(); err == context.Canceled {
```

Comment:
I think you can simply check `err != nil` instead of specifically cancelled. All `ctx.Err() != nil` states are terminal and mean that no further work can be performed.